base_set/cards/mine.py

```diff
@@ -0,0 +1,53 @@
+from core.card import CardType
+from core.card import KingdomCard
+from core.card import TreasureCard
+from core.decision import Decision
+from core.locations import Location
+from core.locations import LocationName
+
+
+class MineTrashDecision(Decision):
```

Comment:
should both the trash and gain happen at the same time? I feel like agents will calculate them at the same time. do they have to hold their state in memory or something and hang until the next decision

Reply:
We could compute every combination of (card_to_trash, card_to_gain) and pass those options. Still, other sets may cause something to happen in between the trash and the gain. Like isn't there some card that says when you trash something you may discard it instead, so we'd have to then insert that decision in between.